### PR TITLE
Added security and refactor/added tests

### PR DESCRIPTION
**Pull request by lemunozm**
### Fix 2 issues with microCDR:
 - Check the max size at deserialization functions (change the API).
 - Better management of the *MicroCDR error* that could generate indeterminate behaviour (change the error type in MicroBuffer struct)

### Tests:
- Refactor tests.
- Added FullBuffer error tests.
- Added SequenceOverflow tests
- Added Endianness tests for all types.
- Improvements in the current tests.

### CODE REVIEW

File: include/microcdr/common.h [hunk 1]
```diff
@@ -55,14 +41,43 @@ typedef struct MicroBuffer
     Endianness endianness;
     uint32_t last_data_size;
 
-    BufferError error;
+    bool error;
```
Comment (richiware) [inline include/microcdr/common.h]: Create a function to get the error flag and another to unset the error flag

Comment (lemunozm) [reply include/microcdr/common.h]: Done!

File: include/microcdr/types/sequence.h [hunk 2]
```diff
@@ -21,64 +21,61 @@ extern "C" {
...
-microcdr_DllAPI bool deserialize_sequence_uint16_t(MicroBuffer* buffer, uint16_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_uint32_t(MicroBuffer* buffer, uint32_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_uint64_t(MicroBuffer* buffer, uint64_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_int8_t(MicroBuffer* buffer, int8_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_int16_t(MicroBuffer* buffer, int16_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_int32_t(MicroBuffer* buffer, int32_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_int64_t(MicroBuffer* buffer, int64_t* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_float(MicroBuffer* buffer, float* array, uint32_t* size);
-microcdr_DllAPI bool deserialize_sequence_double(MicroBuffer* buffer, double* array, uint32_t* size);
+microcdr_DllAPI bool deserialize_sequence_char(MicroBuffer* mb, char* array, uint32_t* size, const uint32_t max_size);
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/microcdr/types/sequence.h]: review the location of array_size

File: test/SequenceOverflow.cpp [hunk 3]
```diff
@@ -44,7 +44,7 @@ TEST_F(SequenceOverflow, Block1)
     uint8_t output[ARRAY_LENGTH];
 
     EXPECT_TRUE(serialize_sequence_uint8_t(&writer, input, SEQUENCE_LENGTH_OVERFLOW));
-    EXPECT_FALSE(deserialize_sequence_uint8_t(&reader, output, &output_size, ARRAY_LENGTH));
+    EXPECT_FALSE(deserialize_sequence_uint8_t(&reader, output, ARRAY_LENGTH, &output_size));
```
Comment (richiware) [inline test/SequenceOverflow.cpp]: For consistence, change ARRAY_LENGTH with ARRAY_CAPACITY

Comment (lemunozm) [reply test/SequenceOverflow.cpp]: Done
Do it also with SEQUENCE_LENGTH to SEQUENCE_SIZE.

**Pull request merged**